EmptyState Migration for Remaining Screens with Logic Alignment and Test Updates

rawadhossain opened the pull request:
<!--
This section can be deleted after reading.

We employ the following branching strategy to simplify the development process and to ensure that only stable code is pushed to the `master` branch:

- `develop`: For unstable code: New features and bug fixes.
- `master`: Where the stable production ready code lies. Only security related bugs.

NOTE!!!

ONLY SUBMIT PRS AGAINST OUR `DEVELOP` BRANCH. THE DEFAULT IS `MAIN`, SO YOU WILL HAVE TO MODIFY THIS BEFORE SUBMITTING YOUR PR FOR REVIEW. PRS MADE AGAINST `MAIN` WILL BE CLOSED.

-->

<!--
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request.
-->

**What kind of change does this PR introduce?**
Refactor specialized screens (`FundCampaignPledge, Pledges, Leaderboard, OrganizationPeople, BlockUser`) with context-aware empty state implementations.

**Issue Number:**
Fixes #5124

**Snapshots/Videos:**

<!--Add snapshots or videos wherever possible.-->

**If relevant, did you update the documentation?**

<!--Add link to Talawa-Docs.-->

**Summary**
• Replaced legacy Stack-based empty views with standardized `EmptyState` component in FundCampaignPledge, UserPortal/Pledges, Leaderboard, OrganizationPeople, and BlockUser screens

• Localized all DataGrid column headers and UI strings, adding new translation keys across all 5 locales (en/es/fr/hi/zh)

• Added/updated tests with empty state assertions, edge cases, and interaction flows for all modified screens

• Fixed invalid variable declarations, added utility classes (flexCenter, dataGridNoHover, etc.), and introduced avatar-related styling tokens in app-fixed.module.css

• Removed debug statements, replaced inline styles with CSS modules, and ensured TypeScript/accessibility standards compliance


**Does this PR introduce a breaking change?**
No

## Checklist

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [ ] I have written tests for all new changes/features
- [ ] I have verified that test coverage meets or exceeds 95%
- [ ] I have run the test suite locally and all tests pass


**Other information**

<!--Add extra information about this PR here-->

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

<!--Yes or No-->


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Leaderboard updates: medal ranks, volunteer hours and email shown; event dashboard visibility control; expanded pledge UI with pledge date, pledged/donated amounts and “more users” indicator
* **Bug Fixes**
  * Consistent empty-state presentation across lists and grids; improved login error verification flow
* **Localization**
  * Added/updated translation keys for leaderboard, pledges, events and common UI in multiple languages
* **Style**
  * Tokenized CSS, avatar styling refactor, reduced inline styles and improved accessibility labels
* **Tests**
  * Added empty-state, localization and ARIA/interaction tests; enhanced end-to-end toast checks

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

rawadhossain commented on src/screens/FundCampaignPledge/FundCampaignPledge.spec.tsx:
```diff
@@ -360,6 +360,36 @@ describe('Testing Campaign Pledge Screen', () => {
+      },
+      { timeout: 3000 },
+    );
+  });
```
In this test we intentionally read from the test i18n fixture directly and allow a fallback value, since the goal here is to verify the rendered UI label rather than the internal i18next namespace resolution. Given this is a controlled test-only i18n setup and the assertion already validates visible behavior, I think the current approach is sufficient.

rawadhossain commented on src/style/app-fixed.module.css:
```diff
@@ -10093,171 +10028,111 @@ form {
-  font-size: 2rem;
+.memberNameFontSize {
+  font-size: 15px;
 }
```
these utilities were added intentionally to reduce duplication and keep layouts consistent.

rawadhossain commented on public/locales/fr/translation.json:
```diff
@@ -9,7 +9,14 @@
     "monthly": "Ce Mois",
     "yearly": "Cette Année",
-    "noVolunteers": "Aucun Bénévole Trouvé!"
+    "noVolunteers": "Aucun Bénévole Trouvé!",
```
the current capitalization is intentional and aligns with the UI tone of this screen, so no change is needed right now.

rawadhossain commented on public/locales/es/common.json:
```diff
@@ -52,7 +52,8 @@
   "joined": "Joined",
-  "joined on": "Joined on",
+  "joined on": "Unido el",
+  "joinedOn": "Unido el",
```
Both joined on and joinedOn are intentionally present across all locale files to support existing usage patterns, so removing one would be a broader i18n refactor and is out of scope for this change. Resolving as-is.

rawadhossain commented:
@coderabbitai full review based on this prompt:
https://github.com/PalisadoesFoundation/talawa-admin/pull/5782#issuecomment-3691518435

rawadhossain commented:
@palisadoes please take a look at this PR add the label. It should be good to go. Thanks.

palisadoes commented:
@coderabbitai full review using this prompt:
- https://github.com/PalisadoesFoundation/talawa-admin/pull/5407#issuecomment-3689679106

palisadoes commented:
1. Please fix the failing e2e test.
1. It may be due to a recently merged API PR, or a feature that you've added that isn't yet supported. 
1. If it's an API issue, week need to fix it quickly as it's affecting multiple other PRs in this repo

palisadoes commented:
Please fix the conflicting files

rawadhossain commented on src/components/EventManagement/EventActionItems/EventActionItems.tsx:
```diff
@@ -250,9 +250,8 @@ const EventActionItems: React.FC<InterfaceEventActionItemsProps> = ({
         return (
           <div
-            className="d-flex fw-bold align-items-center ms-2"
+            className={`d-flex fw-bold align-items-center ms-2 ${styles.assigneeCellContainer}`}
```
The styles.assigneeCellContainer class is already applied correctly and the existing tests focus on verifying functional behavior and rendered output rather than CSS-module implementation details. Adding explicit toHaveClass(styles.assigneeCellContainer) assertions would couple tests to hashed CSS module names without improving coverage or reliability. UI behavior is already fully covered for all assignee scenarios, I don’t see additional value in changing the tests here.

rawadhossain commented on src/screens/Leaderboard/Leaderboard.tsx:
```diff
@@ -173,115 +162,82 @@ function leaderboard(): JSX.Element {
+              />
+            )}
+            {firstName} {lastName}
+          </div>
```
I’ll keep the current implementation. The existing div with role="button", tabIndex, and appropriate ARIA attributes already meets accessibility requirements and is consistent with similar patterns used across the codebase. Switching to a native <button> here would be a stylistic refactor rather than a functional or accessibility fix, so I’m opting not to change it in this PR.

rawadhossain commented on public/locales/zh/translation.json:
```diff
@@ -9,7 +9,14 @@
+    "rank": "排名",
+    "volunteer": "志愿者",
+    "email": "电子邮件",
+    "hoursVolunteered": "志愿服务时数"
```
I’ll keep the current wording as it is. The existing translation is semantically correct and understandable in context, and there’s no functional or consistency issue that warrants changing it at this time. Since this is a trivial wording preference rather than a correctness fix, I’m opting not to modify the translation.

rawadhossain commented on src/screens/Leaderboard/Leaderboard.spec.tsx:
```diff
@@ -269,18 +296,81 @@ describe('Testing Leaderboard Screen', () => {
+    await waitFor(() => {
+      expect(screen.getByTestId('memberScreen')).toBeInTheDocument();
+    });
+  });
```
they intentionally remain separate to clearly document and validate each keyboard interaction path. This improves readability and makes failures easier to diagnose if one key handling regresses.

rawadhossain commented on src/screens/Leaderboard/Leaderboard.spec.tsx:
```diff
@@ -269,18 +296,81 @@ describe('Testing Leaderboard Screen', () => {
+      },
+      { timeout: 2000 },
+    );
+  });
```
his test is intentionally scoped to validating the functional empty-state rendering when search yields no results. ARIA attributes, copy, and iconography are already covered by component level responsibility and existing EmptyState tests.

rawadhossain commented:
While working, the pre-commit hook started failing due to a POM violation in existing Cypress tests. Some login e2e tests were directly using `cy.get('.Toastify__toast')`, which the repo’s `check_pom.js` script explicitly disallows. 

This appears to have been triggered after the recent merge in PR [#5432](https://github.com/PalisadoesFoundation/talawa-admin/pull/5432), which directly used `cy.get('.Toastify__toast')` in login tests.

Changes made to align with the enforced POM rules:
* Added a small helper method `verifyErrorToast()` to `LoginPage`
* Updated the affected Cypress tests to call this POM instead of using `cy.get()` directly
